data-prepper-plugins/dissect-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/dissect/Dissector.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.dissect;
+
+import org.opensearch.dataprepper.plugins.processor.dissect.Fields.AppendField;
+import org.opensearch.dataprepper.plugins.processor.dissect.Fields.Field;
+import org.opensearch.dataprepper.plugins.processor.dissect.Fields.FieldHelper;
+import org.opensearch.dataprepper.plugins.processor.dissect.Fields.IndirectField;
+import org.opensearch.dataprepper.plugins.processor.dissect.Fields.NormalField;
+import org.opensearch.dataprepper.plugins.processor.dissect.Fields.SkipField;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class Dissector {
```

Comment:
Looks like this class has no unit tests and is being tested indirectly in `DissectProcessorTest`. Ideally we would mock this class and have unit tests for it directly, but I won't block the PR on that

Reply:
Makes sense. I separated the tests for Dissector from DissectProcessorTest. `DissectProcessorTest` now only test things after the actual dissection is done - e.g., handle dissect exception, convert type, and put dissected fields into event.